geode-docs/developing/transactions/directed_example.html.md.erb

```diff
@@ -0,0 +1,234 @@
+    commitConflict = false;
+  } 
+  catch (CommitConflictException conflict) {
+    // entry value changed causing a conflict for this customer, so try again
+        commitConflict = true;
+  }
+} while (commitConflict);
+```
+
+Design transactions such that any get operations are within the transaction.
+This causes those entries to be part of the transactional state,
+which is desired such that intersecting transactions can be detected
+and signal commit conficts.
+
+Note that variables `cashBalance` and `tradeBalance` are
+references,
+because the `Region.get` operation returns a reference.
+Read [Region Operations Return References](design_considerations.html#copy-on-read-transactions)
+for details and how to work around the
+implications of a reference as a return value.
+
+## Transaction within a Function
+
+A transaction may be embedded in a function.
+The application invokes the function,
+and the function contains the transaction that does the `begin`,
+the region operations, and the `commit` or `rollback`.
+
+This use of a function can have performance benefits.
+The performance benefit results from both the function
+and the region data residing on servers.
+As the function invokes region operations,
+those operations on region entries stay on the server,
+so there is no network round trip time to do get or put
+operations on region data.
+
+This function example accomplishes atomic updates on a single
+region representing the quantity of products available in inventory.
+Doing this in a transaction prevents double allocating inventory for
+two orders placed simultaneously.
+
+
+``` pre
+/**
+ * Atomically reduce inventory quantity
+ */
+public class TransactionalFunction extends Function {
+
+  /**
+   * Returns true if the function had the requested quantity of
+   * inventory, and successfully completed the transaction to 
+   * record the reduced inventory that fulfills the order.
+   */
+  @Override
+  public void execute(FunctionContext context) {
+    RegionFunctionContext rfc = (RegionFunctionContext) context;
+    Region<ProductId, Integer> inventoryRegion = rfc.getDataSet();
+
+    CacheTransactionManager 
+        mgr = CacheFactory.getAnyInstance().getCacheTransactionManager();
+
+    // single argument will be a ProductId and a quantity
+    ProductRequest request = (ProductRequest) rfc.getArguments();
+    ProductId productRequested = request.getProductId();
+    Integer qtyRequested = request.getQuantity();
+ 
+    Boolean success = false;
+    Boolean commitConflict = false;
+
+    do {
+      try {
+        mgr.begin();
+
+        Integer qtyAvailable = inventoryRegion.get(productRequested);
+        Integer qtyRequested = request.getQuantity();
+        if (qtyAvail >= qtyRequested) {
+          // enough inventory is available, so process request
+          Integer remaining = qtyAvailable - qtyRequested;
+          inventoryRegion.put(productRequested, remaining);
+          success = true;
+        } 
```

Comment:
Possibly should add handle of other TransactionExceptions here to avoid dangling txState here.
catch (TransactionException exception) {
success = false;
mgr.rollback();
break;
}